src/megatron/bridge/models/model_provider.py

```diff
@@ -460,14 +456,18 @@ def get_model(
     else:
         model = _create_model(model_provider, model_type)
 
-    if pre_wrap_hook:
-        _model = pre_wrap_hook(model)
-        if _model is not None:
-            model = _model
-
```

Comment:
why does this need to be changed? this will break peft here: https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/2f18dff6971b977454de128cde338efcff338c33/src/megatron/bridge/training/setup.py#L326-L370

since this will now call load_checkpoint once per model chunk instead of once overall before applying peft over the chunks

Reply:
i will change in rl side then. I will just update to allow the hook to be a list of callables.

Reply:
the user can call `model_provider.register_pre_wrap_hook(...)` multiple times:
https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/46c79c67b318f1aa73e279e3ab0c01d5c57f8b23/src/megatron/bridge/models/model_provider.py#L261C18-L261C33

which will be composed here: https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/46c79c67b318f1aa73e279e3ab0c01d5c57f8b23/src/megatron/bridge/models/model_provider.py#L217-L239